shared/manifest.go

```diff
@@ -154,15 +231,25 @@ func ExplodePossibleFilenames(filePath string) []string {
 	return nil
 }
 
-func trieContains(t interface{}, parts []string) bool {
-	if len(parts) == 0 {
-		return t != nil
+// ParseTestURL parses a WPT test URL and returns its file path and query
+// components. If the test is a multi-global (auto-generated) test, the
+// function returns the underlying file name of the test.
+// e.g. testURL="foo/bar/test.any.worker.html?varaint"
+//      filepath="foo/bar/test.any.js"
+//      query="?variant"
+func ParseTestURL(testURL string) (filePath, query string) {
+	filePath = testURL
+	if qPos := strings.Index(testURL, "?"); qPos > -1 {
+		filePath = testURL[:qPos]
+		query = testURL[qPos:]
 	}
-
-	// t could be nil (e.g. if the previous part does not exist in the map), in which case casting will fail.
-	trie, ok := t.(map[string]interface{})
-	if !ok {
-		return false
+	for _, i := range implosions() {
+		tSuffix := i[0]
```

Comment:
Eugh, I hate golang. Would it be so dreadful to name this `testSuffix` and `fileSuffix`?

Reply:
They'd be `ts` and `fs` in true Golang style. :)

And no personally I don't think it's worthwhile to save a few keystrokes, either...